externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetPrimaryKeys.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.flink.operation
+
+import scala.collection.JavaConverters._
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.flink.table.api.{DataTypes, ResultKind}
+import org.apache.flink.table.catalog.Column
+import org.apache.flink.types.Row
+
+import org.apache.kyuubi.engine.flink.result.ResultSet
+import org.apache.kyuubi.operation.meta.ResultSetSchemaConstant._
+import org.apache.kyuubi.session.Session
+
+class GetPrimaryKeys(
+    session: Session,
+    catalogNameOrEmpty: String,
+    schemaNameOrEmpty: String,
+    tableName: String)
+  extends FlinkOperation(session) {
+
+  override protected def runInternal(): Unit = {
+    try {
+      val tableEnv = sessionContext.getExecutionContext.getTableEnvironment
+
+      val catalogName =
+        if (StringUtils.isEmpty(catalogNameOrEmpty)) tableEnv.getCurrentCatalog
+        else catalogNameOrEmpty
+
+      val schemaName =
+        if (StringUtils.isEmpty(schemaNameOrEmpty)) tableEnv.getCurrentDatabase
+        else schemaNameOrEmpty
```

Comment:
It seems that there is still a problem here. When the catalogName is not equal to tableEnv.getCurrentCatalog, the obtained schemaName is still the CurrentDatabase of the current catalog. Correctly modify the code should be similar to the following way.


```val schemaName =
        if (StringUtils.isEmpty(schemaNameOrEmpty)) {
          if (catalogName != tableEnv.getCurrentCatalog) {
            catalog.getDefaultDatabase
          } else {
            tableEnv.getCurrentDatabase
          }
        } else schemaNameOrEmpty

Reply:
val schemaName =
        if (StringUtils.isEmpty(schemaNameOrEmpty)) {
          if (tableEnv.getCurrentDatabase != null) {
            tableEnv.getCurrentDatabase
          } else {
            tableEnv.getCatalog(catalogName).get().getDefaultDatabase
          }
        } else schemaNameOrEmpty  Is that OK?

Reply:
Thank you for your reviews. I think it's good. I have revised the relevant content.

Reply:
Can we add a test for this logic?